lib/src/rules/overridden_fields.dart

```diff
@@ -162,7 +162,7 @@ class _Visitor extends SimpleAstVisitor<void> {
 
     Iterable<InterfaceType> interfaces;
     if (classElement is MixinElement) {
-      interfaces = _findAllSupertypesInMixin(classElement);
+      interfaces = _findAllSupertypesInMixin(classElement as MixinElement);
```

Comment:
I believe that line 158 is wrong and needs to test for `InterfaceElement` (or maybe this line needs to test `enclosingElement is MixinElement` and skip the local variable `classElement`). As it is, I think that `classElement is MixinElement` will always be false.

Also, if the else block is dependent on `classElement` being a `ClassElement` then you'll need to add a test on line 166.

Reply:
I think it should be `InterfaceElement`.

Updated.